include/deal.II/fe/fe_point_evaluation.h

```diff
@@ -530,6 +672,163 @@ FEPointEvaluation<n_components, dim, spacedim>::evaluate(
     }
 }
 
+
+
+template <int n_components, int dim, int spacedim>
+void
+FEPointEvaluation<n_components, dim, spacedim>::integrate(
+  const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+  const ArrayView<const Point<dim>> &                         unit_points,
+  const ArrayView<double> &                                   solution_values,
+  const ArrayView<const value_type> &                         values,
+  const ArrayView<const gradient_type> &                      gradients)
+{
+  AssertDimension(solution_values.size(), fe->dofs_per_cell);
+  if (false /*TODO*/ &&
+      ((values.size() > 0 || gradients.size() > 0) && !poly.empty()))
+    {
+      // fast path with tensor product integration
+
+      if (solution_renumbered.size() != dofs_per_component)
+        solution_renumbered.resize(dofs_per_component);
+
+      // let mapping compute the transformation
+      if (gradients.size() > 0)
+        {
+          Assert(mapping_q_generic != nullptr, ExcInternalError());
+
+          // cast constness away (input is overriden)
+          ArrayView<gradient_type> gradients_non_const(
+            const_cast<gradient_type *>(gradients.data()), gradients.size());
+          mapping_q_generic->transform_variable(
```

Comment:
I think we need here the transposed of this functions, don't we?

Reply:
Yes, we need to apply the transpose here.